mifosng-android/src/main/java/com/mifos/api/local/databasehelper/DatabaseHelperGroups.java

```diff
@@ -34,50 +38,134 @@ public DatabaseHelperGroups() {
-     * insert.
+     * This Method Saving the Single Group in the Database
      *
-     * @param groupPage
-     * @return null
+     * @param group
+     * @return Observable.just(Group)
      */
-    @Nullable
-    public Observable<Void> saveGroups(final Page<Group> groupPage) {
-        AsyncTask.THREAD_POOL_EXECUTOR.execute(new Runnable() {
+    public Observable<Group> saveGroup(final Group group) {
+        return Observable.defer(new Func0<Observable<Group>>() {
             @Override
-            public void run() {
+            public Observable<Group> call() {
 
-                for (Group group : groupPage.getPageItems()) {
-                    group.save();
-                }
+                GroupDate groupDate = new GroupDate(group.getId(), 0,
+                        group.getActivationDate().get(0),
+                        group.getActivationDate().get(1),
+                        group.getActivationDate().get(2));
+                group.setGroupDate(groupDate);
+                group.save();
+                return Observable.just(group);
             }
         });
-        return null;
     }
 
-
     /**
-     * Reading All groups from table of Group and return the GroupList
+     * Reading All groups from Database table of Group and return the GroupList
      *
      * @return List Of Groups
      */
-    //TODO Implement Observable Transaction to load Client List
     public Observable<Page<Group>> readAllGroups() {
-        return Observable.create(new Observable.OnSubscribe<Page<Group>>() {
+        return Observable.defer(new Func0<Observable<Page<Group>>>() {
             @Override
-            public void call(Subscriber<? super Page<Group>> subscriber) {
-
+            public Observable<Page<Group>> call() {
                 Page<Group> groupPage = new Page<>();
                 groupPage.setPageItems(SQLite.select()
                         .from(Group.class)
                         .queryList());
-                subscriber.onNext(groupPage);
-                subscriber.onCompleted();
+                return Observable.just(groupPage);
             }
         });
+    }
 
+    /**
+     * This Method Retrieving the Group from the Local Database.
+     *
+     * @param groupId Group Id
+     * @return Group
+     */
+    public Observable<Group> getGroup(final int groupId) {
+        return Observable.defer(new Func0<Observable<Group>>() {
+            @Override
+            public Observable<Group> call() {
+
+                Group group = SQLite.select()
+                        .from(Group.class)
+                        .where(Group_Table.id.eq(groupId))
+                        .querySingle();
+
+                if (group != null) {
+                    group.setActivationDate(Arrays.asList(group.getGroupDate().getDay(),
+                            group.getGroupDate().getMonth(), group.getGroupDate().getYear()));
+                }
+
+                return null;
```

Comment:
Sorry for that When I have written this method, demo Server was down, So I have not test this method, Now fixed.